shell/client/grainview.js

```diff
@@ -401,9 +410,9 @@ GrainView.prototype.sessionStatus = function () {
 GrainView.prototype.route = function () {
   this._dep.depend();
   if (this._token) {
-    return "/shared/" + this._token;
+    return "/shared/" + this._token + this._path;
```

Comment:
Hm. We could include the original query and the original fragment. But what if someone calls setPath() with a new query and a new fragment? I don't think we've nailed down exactly the semantics we want there. Should we ignore any query and fragment that comes in on setPath()?